jax/_src/numpy/ufuncs.py

```diff
@@ -2553,11 +2553,14 @@ def divmod(x1: ArrayLike, x2: ArrayLike, /) -> tuple[Array, Array]:
 def _float_divmod(x1: ArrayLike, x2: ArrayLike) -> tuple[Array, Array]:
   # see float_divmod in floatobject.c of CPython
   mod = lax.rem(x1, x2)
+  mod = _where(logical_and(abs(x2) == np.inf, abs(x1) != np.inf), x1, mod)
   div = lax.div(lax.sub(x1, mod), x2)
 
   ind = lax.bitwise_and(mod != 0, lax.sign(x2) != lax.sign(mod))
   mod = lax.select(ind, mod + x2, mod)
   div = lax.select(ind, div - _constant_like(div, 1), div)
+  if dtypes.dtype(x1) != dtypes.bfloat16:
```

Comment:
This is surprising – can you comment on why we need specialized logic for bfloat16 here?